pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -49,8 +52,44 @@ type RemoteError struct {
 }
 
 // New returns an authenticator which will make a basic auth call to the given url.
-func New(providerName string, url string, mapper authapi.UserIdentityMapper) authenticator.Password {
-	return &Authenticator{providerName, url, mapper}
+func New(providerName string, url string, client *http.Client, mapper authapi.UserIdentityMapper) authenticator.Password {
```

Comment:
I really don't want to make many (any?) other layers depend on our config objects